### PR TITLE
`Assessment`: Fix complaints not accessible after lock runs out

### LINKED ISSUE #5847: Locks for the assessment dashboard cause problems
### Describe the bug

When setting locks in the assessment dashboard for complaints or more feedback requests (which automatically happens when you open a request) they are supposed to be unlocked again automatically after 24 hours. The problem is, that the locks doesn’t get removed but the people who set the lock in the first place can’t work on the case either. That basically means, that the regarding case is completely locked for everyone.

### To Reproduce

1. Open on of the requests
2. The lock will be set automatically
3. Do not resolve the case within 24 hours
4. Do not manually remove the lock 

### Expected behavior

As described above, the lock should be removed automatically or the person holding lock should at least be able to remove it or work on the request.

### Screenshots

Bildschirmfoto 2022-11-16 um 13 43 54

Screenshot taken on 16.10.

### What browsers are you seeing the problem on?

Safari

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
### Motivation, Context and Description

Fix #5847

### Steps for Testing

Prerequisites:
- 3 Tutors
- 1 Student
- 1 Exercise with Complaints enabled

Best tested locally (set COMPLAINT_LOCK_DURATION_IN_MINUTES in Constants.java to 5 minutes)
1. Log in to Artemis
2. Make submission as student
3. Assess the submission as tutor 1
4. Complain as student
5. Lock the complaint as tutor 2
6. Wait until the lock should run out (normally 24h, with local change sooner)
7. Check that you can now assess the complaint as tutor 3

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by krusche — CHANGES_REQUESTED** [review]
please add test case for the 2nd issue that the same tutor cannot finish the complaint after 24h

File: src/test/javascript/spec/component/complaints/complaint-response.service.spec.ts [hunk 1]
```diff
@@ -67,6 +67,12 @@ describe('ComplaintResponseService', () => {
         expect(isLocked).toBeTrue();
     });
 
+    it('should correctly calculate that complaint response is locked by other user', () => {
+        const lockedComplaintResponse = setupLockTest('test', false, 'test2', false);
+        const isLocked = complaintResponseService.isComplaintResponseLockedForLoggedInUser(lockedComplaintResponse, new TextExercise(undefined, undefined));
+        expect(isLocked).toBeFalse();
```
Comment (krusche) [inline src/test/javascript/spec/component/complaints/complaint-response.service.spec.ts:73]: double check that this test actually fails on `develop` and that it covers enough `expect` statements to verify that the functionality is really implemented correctly

Comment (JohannesStoehr) [reply src/test/javascript/spec/component/complaints/complaint-response.service.spec.ts:73]: This actually fails 👍

**Review by manuelmanso — APPROVED** [review]
Tested locally, lock gets released correctly and removing lock manually also works. Code and tests look good too!

One (separate) thing: this alert is shown to the first tutor, should this be an alert? It seems more like something that should be shown in the page itself instead of as an alert.
image

**Review by stefanwaldhauser — APPROVED** [review]
Good job! Fixed the bug!

**Review by tobias-lippert — APPROVED** [review]
Tested locally. Bug has been fixed.

**Review by akesfeden — APPROVED** [review]
Tested locally, locks are released after the time out duration. Also, instructors and admins can override them even if the tutor's lock is not timed out yet.

**Review by krusche — APPROVED** [review]

**Pull request merged**